@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -422,23 +422,46 @@ function placePointLabelChoosingAnchor(
     };
     // Placements options will be read from label.layoutStyle.placements in final solution.
     const placements = possibleAnchorPlacementsCW;
-    // Find current anchor placement on the optional placements list.
-    const idx = placements.findIndex(val => {
-        return val.h === basePlacement.h && val.v === basePlacement.v;
-    });
-    // If not listed in placements list (idx === -1) will start from current anchor setup.
     const placementsNum = placements.length;
+    // Find current anchor placement on the optional placements list or best matching position.
+    // Index of exact match.
+    let matchIdx = -1;
+    // Index of closest placement to the base one.
+    let closeIdxOff = 0;
+    let closeDiff = Infinity;
+    for (let i = 0; i < placementsNum; ++i) {
+        const placement = placements[i];
+        const hDiff = placement.h - basePlacement.h;
+        const vDiff = placement.v - basePlacement.v;
+        const manhattanDist = Math.abs(hDiff) + Math.abs(vDiff);
+        if (manhattanDist === 0) {
+            matchIdx = i;
+            closeIdxOff = 0;
+            break;
+        } else if (manhattanDist < closeDiff) {
+            closeDiff = manhattanDist;
+            closeIdxOff = i;
+        }
+    }
```

Comment:
Frankly, this is overcomplicated. What if we just add the 8 placements in the placement list, but we just test 4 by jumping in increments of 2?
That way if base placement is {Top, Center}, we test  {Top, Center}, {Center,Right}. {Bottom,Center}, {Center,Left}
For base placement is on a corner like {Top,Right}, we test {T,R}, {B,R}, {B,L}, {T,L}.

And you can find always the match simply by doing findIndex(). No more handling of idx === -1 needed. Since {Center,Center} we handle it in placePointLabel.

Reply:
Yes of course this is kind a nice workaround, but I will loose the flexibility and possibility to simply jump to Theme based placements (currently it enough to change array reference).
My proposal which should solve both problems is to provide two alternative tables:
- with centered anchors (North, South, East, West)
- with corner anchors (NW, NE, SE, SW).
Algorithm will check if theme placement is center based or corner and then do finding the closest match and iterate.
Although the current implementation may look over-complicated it is just finding the closest match and best match at the same time, I am doing so just to reduce number of searches (iterations) on placement array.